core/src/test/java/io/cucumber/core/runner/CachingGlueTest.java

```diff
@@ -458,6 +463,20 @@ void scenario_scoped_hooks_have_higher_order() {
 
         assertThat(hooks, contains(hookDefinition2, hookDefinition1, hookDefinition3));
     }
+    
+	@Test
+	public void emits_hook_messages_to_bus() {
+		SpyMessageEnvelopBus spyBus = new SpyMessageEnvelopBus();
```

Comment:
You can use a real bus and add a listener as a lambda in the test.

Reply:
Changed it. Thanks.